generator/internal/http_option_utils.h

```diff
@@ -69,6 +69,24 @@ void SetHttpDerivedMethodVars(
     google::protobuf::MethodDescriptor const& method,
     VarsDictionary& method_vars);
 
+struct QueryParameterInfo {
+  protobuf::FieldDescriptor::CppType cpp_type;
+  // A code fragment the generator emits to access the value of the field.
+  std::string request_field_accessor;
+  // Check presence for MESSAGE types as their default values may result in
+  // undesired behavior.
+  bool check_presence;
```

Comment:
TIL that nested protobuf messages are inherently optional and that: `"" != "foo { }"`

for:
```proto
message Foo {}

message Bar {
  Foo foo = 1;
}
```